`tests/test_project.py::test_user_args` fails when run outside the git repo

When running the test suite unpacked from the git archive, I get the following test failure:

```pytb
nox > Running session test-3.11
nox > Creating virtual environment (virtualenv) using python3.11 in .nox/test-3-11
nox > python -m pip install '.[test]'
nox > pytest --showlocals -vv --cov --cov-report=html:/tmp/meson-python-0.13.0/.nox/test-3-11/htmlcov --cov-report=xml:/tmp/meson-python-0.13.0/.nox/test-3-11/coverage-3.11.xml
========================================================= test session starts =========================================================
platform linux -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0 -- /tmp/meson-python-0.13.0/.nox/test-3-11/bin/python
cachedir: .pytest_cache
rootdir: /tmp/meson-python-0.13.0
configfile: pyproject.toml
testpaths: tests
plugins: cov-4.0.0, mock-3.10.0
collected 88 items                                                                                                                    

tests/test_consistency.py::test_pyproject_dependencies PASSED                                                                   [  1%]
tests/test_editable.py::test_walk PASSED                                                                                        [  2%]
tests/test_editable.py::test_nodes_tree PASSED                                                                                  [  3%]
tests/test_editable.py::test_collect PASSED                                                                                     [  4%]
tests/test_editable.py::test_mesonpy_meta_finder PASSED                                                                         [  5%]
tests/test_editable.py::test_mesonpy_traversable PASSED                                                                         [  6%]
tests/test_editable.py::test_resources PASSED                                                                                   [  7%]
tests/test_editable.py::test_importlib_resources PASSED                                                                         [  9%]
tests/test_editable.py::test_editable_install PASSED                                                                            [ 10%]
tests/test_editable.py::test_editble_reentrant PASSED                                                                           [ 11%]
tests/test_examples.py::test_spam PASSED                                                                                        [ 12%]
tests/test_metadata.py::test_no_pep621 PASSED                                                                                   [ 13%]
tests/test_metadata.py::test_pep621 PASSED                                                                                      [ 14%]
tests/test_metadata.py::test_dynamic_version PASSED                                                                             [ 15%]
tests/test_output.py::test_colors[False-env0-False] PASSED                                                                      [ 17%]
tests/test_output.py::test_colors[True-env1-True] PASSED                                                                        [ 18%]
tests/test_output.py::test_colors[False-env2-False] PASSED                                                                      [ 19%]
tests/test_output.py::test_colors[True-env3-False] PASSED                                                                       [ 20%]
tests/test_output.py::test_colors[False-env4-True] PASSED                                                                       [ 21%]
tests/test_output.py::test_colors[True-env5-True] PASSED                                                                        [ 22%]
tests/test_output.py::test_colors_conflict PASSED                                                                               [ 23%]
tests/test_pep517.py::test_get_requires_for_build_wheel[noninja-patchelf-pure] PASSED                                           [ 25%]
tests/test_pep517.py::test_get_requires_for_build_wheel[noninja-patchelf-library] PASSED                                        [ 26%]
tests/test_pep517.py::test_get_requires_for_build_wheel[noninja-nopatchelf-pure] PASSED                                         [ 27%]
tests/test_pep517.py::test_get_requires_for_build_wheel[noninja-nopatchelf-library] PASSED                                      [ 28%]
tests/test_pep517.py::test_get_requires_for_build_wheel[oldninja-patchelf-pure] PASSED                                          [ 29%]
tests/test_pep517.py::test_get_requires_for_build_wheel[oldninja-patchelf-library] PASSED                                       [ 30%]
tests/test_pep517.py::test_get_requires_for_build_wheel[oldninja-nopatchelf-pure] PASSED                                        [ 31%]
tests/test_pep517.py::test_get_requires_for_build_wheel[oldninja-nopatchelf-library] PASSED                                     [ 32%]
tests/test_pep517.py::test_get_requires_for_build_wheel[newninja-patchelf-pure] PASSED                                          [ 34%]
tests/test_pep517.py::test_get_requires_for_build_wheel[newninja-patchelf-library] PASSED                                       [ 35%]
tests/test_pep517.py::test_get_requires_for_build_wheel[newninja-nopatchelf-pure] PASSED                                        [ 36%]
tests/test_pep517.py::test_get_requires_for_build_wheel[newninja-nopatchelf-library] PASSED                                     [ 37%]
tests/test_pep517.py::test_invalid_config_settings PASSED                                                                       [ 38%]
tests/test_pep517.py::test_invalid_config_settings_suggest PASSED                                                               [ 39%]
tests/test_pep517.py::test_validate_config_settings_invalid PASSED                                                              [ 40%]
tests/test_pep517.py::test_validate_config_settings_repeated PASSED                                                             [ 42%]
tests/test_pep517.py::test_validate_config_settings_str PASSED                                                                  [ 43%]
tests/test_pep517.py::test_validate_config_settings_list PASSED                                                                 [ 44%]
tests/test_pep517.py::test_validate_config_settings_tuple PASSED                                                                [ 45%]
tests/test_pep518.py::test_pep518[sdist_to_wheel-purelib-and-platlib] PASSED                                                    [ 46%]
tests/test_pep518.py::test_pep518[wheel_directly-purelib-and-platlib] PASSED                                                    [ 47%]
tests/test_project.py::test_name[library] PASSED                                                                                [ 48%]
tests/test_project.py::test_name[library-pep621] PASSED                                                                         [ 50%]
tests/test_project.py::test_version[library] PASSED                                                                             [ 51%]
tests/test_project.py::test_version[library-pep621] PASSED                                                                      [ 52%]
tests/test_project.py::test_unsupported_dynamic PASSED                                                                          [ 53%]
tests/test_project.py::test_unsupported_python_version PASSED                                                                   [ 54%]
tests/test_project.py::test_user_args FAILED                                                                                    [ 55%]
tests/test_project.py::test_unknown_user_args[top-level] PASSED                                                                 [ 56%]
tests/test_project.py::test_unknown_user_args[meson-args] PASSED                                                                [ 57%]
tests/test_project.py::test_install_tags PASSED                                                                                 [ 59%]
tests/test_project.py::test_validate_pyproject_config_one PASSED                                                                [ 60%]
tests/test_project.py::test_validate_pyproject_config_all PASSED                                                                [ 61%]
tests/test_project.py::test_validate_pyproject_config_unknown PASSED                                                            [ 62%]
tests/test_project.py::test_validate_pyproject_config_empty PASSED                                                              [ 63%]
tests/test_project.py::test_invalid_build_dir PASSED                                                                            [ 64%]
tests/test_project.py::test_compiler SKIPPED (Requires MSVC)                                                                    [ 65%]
tests/test_sdist.py::test_contents PASSED                                                                                       [ 67%]
tests/test_sdist.py::test_contents_subdirs PASSED                                                                               [ 68%]
tests/test_sdist.py::test_contents_unstaged PASSED                                                                              [ 69%]
tests/test_sdist.py::test_executable_bit PASSED                                                                                 [ 70%]
tests/test_sdist.py::test_generated_files PASSED                                                                                [ 71%]
tests/test_tags.py::test_wheel_tag PASSED                                                                                       [ 72%]
tests/test_tags.py::test_macos_platform_tag SKIPPED (macOS specific test)                                                       [ 73%]
tests/test_tags.py::test_python_host_platform SKIPPED (macOS specific test)                                                     [ 75%]
tests/test_tags.py::test_tag_empty_wheel PASSED                                                                                 [ 76%]
tests/test_tags.py::test_tag_purelib_wheel PASSED                                                                               [ 77%]
tests/test_tags.py::test_tag_platlib_wheel PASSED                                                                               [ 78%]
tests/test_tags.py::test_tag_stable_abi PASSED                                                                                  [ 79%]
tests/test_tags.py::test_tag_mixed_abi PASSED                                                                                   [ 80%]
tests/test_wheel.py::test_scipy_like PASSED                                                                                     [ 81%]
tests/test_wheel.py::test_contents PASSED                                                                                       [ 82%]
tests/test_wheel.py::test_purelib_and_platlib PASSED                                                                            [ 84%]
tests/test_wheel.py::test_pure PASSED                                                                                           [ 85%]
tests/test_wheel.py::test_configure_data PASSED                                                                                 [ 86%]
tests/test_wheel.py::test_local_lib PASSED                                                                                      [ 87%]
tests/test_wheel.py::test_contents_license_file PASSED                                                                          [ 88%]
tests/test_wheel.py::test_executable_bit PASSED                                                                                 [ 89%]
tests/test_wheel.py::test_detect_wheel_tag_module PASSED                                                                        [ 90%]
tests/test_wheel.py::test_detect_wheel_tag_script PASSED                                                                        [ 92%]
tests/test_wheel.py::test_rpath PASSED                                                                                          [ 93%]
tests/test_wheel.py::test_uneeded_rpath PASSED                                                                                  [ 94%]
tests/test_wheel.py::test_entrypoints PASSED                                                                                    [ 95%]
tests/test_wheel.py::test_top_level_modules PASSED                                                                              [ 96%]
tests/test_wheel.py::test_purelib_platlib_split PASSED                                                                          [ 97%]
tests/test_wheelfile.py::test_basic PASSED                                                                                      [ 98%]
tests/test_wheelfile.py::test_source_date_epoch PASSED                                                                          [100%]/tmp/meson-python-0.13.0/.nox/test-3-11/lib/python3.11/site-packages/coverage/html.py:116: CoverageWarning: No contexts were measured
  self.coverage._warn("No contexts were measured")


============================================================== FAILURES ===============================================================
___________________________________________________________ test_user_args ____________________________________________________________

package_user_args = PosixPath('/tmp/meson-python-0.13.0/tests/packages/user-args')
tmp_path = PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fab2c60fe90>

    def test_user_args(package_user_args, tmp_path, monkeypatch):
        project_run = mesonpy.Project._run
        cmds = []
        args = []
    
        def wrapper(self, cmd):
            # intercept and filter out test arguments and forward the call
            if cmd[:2] == ['meson', 'compile']:
                # when using meson compile instead of ninja directly, the
                # arguments needs to be unmarshalled from the form used to
                # pass them to the --ninja-args option
                assert cmd[-1].startswith('--ninja-args=')
                cmds.append(cmd[:2])
                args.append(ast.literal_eval(cmd[-1].split('=')[1]))
            elif cmd[:1] == ['meson']:
                cmds.append(cmd[:2])
                args.append(cmd[2:])
            else:
                # direct ninja invocation
                cmds.append([os.path.basename(cmd[0])])
                args.append(cmd[1:])
            return project_run(self, [x for x in cmd if not x.startswith(('config-', 'cli-', '--ninja-args'))])
    
        monkeypatch.setattr(mesonpy.Project, '_run', wrapper)
    
        config_settings = {
            'dist-args': ('cli-dist',),
            'setup-args': ('cli-setup',),
            'compile-args': ('cli-compile',),
            'install-args': ('cli-install',),
        }
    
>       mesonpy.build_sdist(tmp_path, config_settings)

args       = [['/tmp/meson-python-0.13.0/tests/packages/user-args',
  '/tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build',
  '-Dbuildtype=release',
  '-Db_ndebug=if-release',
  '-Db_vscrt=md',
  'config-setup',
  'cli-setup',
  '--native-file=/tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build/meson-python-native-file.ini'],
 ['--allow-dirty',
  '--no-tests',
  '--formats',
  'gztar',
  'config-dist',
  'cli-dist']]
cmds       = [['meson', 'setup'], ['meson', 'dist']]
config_settings = {'compile-args': ('cli-compile',),
 'dist-args': ('cli-dist',),
 'install-args': ('cli-install',),
 'setup-args': ('cli-setup',)}
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7fab2c60fe90>
package_user_args = PosixPath('/tmp/meson-python-0.13.0/tests/packages/user-args')
project_run = <function Project._run at 0x7fab2c5aec00>
tmp_path   = PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0')
wrapper    = <function test_user_args.<locals>.wrapper at 0x7fab2c5e5bc0>

/tmp/meson-python-0.13.0/tests/test_project.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/tmp/meson-python-0.13.0/mesonpy/__init__.py:1046: in wrapper
    return func(*args, **kwargs)
        args       = (PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0'),
 {'compile-args': ('cli-compile',),
  'dist-args': ('cli-dist',),
  'install-args': ('cli-install',),
  'setup-args': ('cli-setup',)})
        func       = <function build_sdist at 0x7fab2c57f100>
        kwargs     = {}
/tmp/meson-python-0.13.0/mesonpy/__init__.py:1072: in build_sdist
    return project.sdist(out).name
        config_settings = {'compile-args': ('cli-compile',),
 'dist-args': ('cli-dist',),
 'install-args': ('cli-install',),
 'setup-args': ('cli-setup',)}
        out        = PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0')
        project    = <mesonpy.Project object at 0x7fab2c5fccd0>
        sdist_directory = PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0')
/tmp/meson-python-0.13.0/mesonpy/__init__.py:908: in sdist
    self._run(['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar', *self._meson_args['dist']])
        directory  = PosixPath('/tmp/pytest-of-mgorny/pytest-2/test_user_args0')
        self       = <mesonpy.Project object at 0x7fab2c5fccd0>
/tmp/meson-python-0.13.0/tests/test_project.py:84: in wrapper
    return project_run(self, [x for x in cmd if not x.startswith(('config-', 'cli-', '--ninja-args'))])
        args       = [['/tmp/meson-python-0.13.0/tests/packages/user-args',
  '/tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build',
  '-Dbuildtype=release',
  '-Db_ndebug=if-release',
  '-Db_vscrt=md',
  'config-setup',
  'cli-setup',
  '--native-file=/tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build/meson-python-native-file.ini'],
 ['--allow-dirty',
  '--no-tests',
  '--formats',
  'gztar',
  'config-dist',
  'cli-dist']]
        cmd        = ['meson',
 'dist',
 '--allow-dirty',
 '--no-tests',
 '--formats',
 'gztar',
 'config-dist',
 'cli-dist']
        cmds       = [['meson', 'setup'], ['meson', 'dist']]
        project_run = <function Project._run at 0x7fab2c5aec00>
        self       = <mesonpy.Project object at 0x7fab2c5fccd0>
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <mesonpy.Project object at 0x7fab2c5fccd0>, cmd = ['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar']

    def _run(self, cmd: Sequence[str]) -> None:
        """Invoke a subprocess."""
        print('{cyan}{bold}+ {}{reset}'.format(' '.join(cmd), **_STYLES))
        r = subprocess.run(cmd, env=self._env, cwd=self._build_dir)
        if r.returncode != 0:
>           raise SystemExit(r.returncode)
E           SystemExit: 1

cmd        = ['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar']
r          = CompletedProcess(args=['meson', 'dist', '--allow-dirty', '--no-tests', '--formats', 'gztar'], returncode=1)
self       = <mesonpy.Project object at 0x7fab2c5fccd0>

/tmp/meson-python-0.13.0/mesonpy/__init__.py:739: SystemExit
-------------------------------------------------------- Captured stdout call ---------------------------------------------------------
+ meson setup /tmp/meson-python-0.13.0/tests/packages/user-args /tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=/tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build/meson-python-native-file.ini
The Meson build system
Version: 1.1.0
Source dir: /tmp/meson-python-0.13.0/tests/packages/user-args
Build dir: /tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build
Build type: native build
Project name: user-args
Project version: 1.0.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program python found: YES (/tmp/meson-python-0.13.0/.nox/test-3-11/bin/python)
Build targets in project: 0

user-args 1.0.0

  User defined options
    Native files: /tmp/meson-python-0.13.0/tests/packages/user-args/.mesonpy-y5xxqbns/build/meson-python-native-file.ini
    buildtype   : release
    b_ndebug    : if-release
    b_vscrt     : md

Found ninja-1.11.1 at /usr/bin/ninja
+ meson dist --allow-dirty --no-tests --formats gztar
Dist currently only works with Git or Mercurial repos

---------- coverage: platform linux, python 3.11.3-final-0 -----------
Coverage HTML written to dir /tmp/meson-python-0.13.0/.nox/test-3-11/htmlcov
Coverage XML written to file /tmp/meson-python-0.13.0/.nox/test-3-11/coverage-3.11.xml

======================================================= short test summary info =======================================================
SKIPPED [1] tests/test_project.py:224: Requires MSVC
SKIPPED [1] tests/test_tags.py:36: macOS specific test
SKIPPED [1] tests/test_tags.py:47: macOS specific test
FAILED tests/test_project.py::test_user_args - SystemExit: 1
========================================= 1 failed, 84 passed, 3 skipped in 253.81s (0:04:13) =========================================
nox > Command pytest --showlocals -vv --cov --cov-report=html:/tmp/meson-python-0.13.0/.nox/test-3-11/htmlcov --cov-report=xml:/tmp/meson-python-0.13.0/.nox/test-3-11/coverage-3.11.xml failed with exit code 1
nox > Session test-3.11 failed.
```

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -2,7 +2,6 @@
 #
 # SPDX-License-Identifier: MIT
 
-import atexit
 import contextlib
 import os
 import os.path
@@ -73,24 +72,6 @@ def tmp_path_session(tmp_path_factory):
         prefix='mesonpy-test-',
         dir=tmp_path_factory.mktemp('test'),
     ))
-
-
-@pytest.fixture
-def tmp_build_path():
-    # Similar to the tmp_path fixture but ensures that the temporary
-    # directory is on the same Windows drive as the test packages
-    # source directory. This is required to work around a Cython bug
-    # exposed in our test packages. See https://github.com/cython/cython/pull/5279
-    tmpdir = tempfile.mkdtemp(prefix='.build-', dir=package_dir)
-    # The clenup is implemented with an atexit hook rather than with a
-    # regular context manager because we import extension modules from
-    # the build directory in the tests and these need to stay around
-    # till the Python interpreter is being teared down. Orherwise
-    # PermissionError removing the folder are obtained on Windows, but
-    # more importantly, the removal succeeds on Cygwin but memory
-    # mapping errors are obtained when Python tries to fork().
-    atexit.register(shutil.rmtree, tmpdir, ignore_errors=True)
-    yield pathlib.Path(tmpdir)
 
 
 class VEnv(EnvBuilder):

--- a/tests/test_editable.py
+++ b/tests/test_editable.py
@@ -58,10 +58,10 @@ def test_collect(package_complex):
     assert tree['complex']['more']['__init__.py'] == os.path.join(root, 'complex', 'more', '__init__.py')
 
 
-def test_mesonpy_meta_finder(package_complex, tmp_build_path):
+def test_mesonpy_meta_finder(package_complex, tmp_path):
     # build a package in a temporary directory
-    mesonpy.Project(package_complex, tmp_build_path)
-    build_path = tmp_build_path / 'build'
+    mesonpy.Project(package_complex, tmp_path)
+    build_path = tmp_path / 'build'
 
     # point the meta finder to the build directory
     finder = _editable.MesonpyMetaFinder({'complex'}, os.fspath(build_path), ['ninja'])

--- a/tests/test_project.py
+++ b/tests/test_project.py
@@ -19,7 +19,7 @@ import pytest
 
 import mesonpy
 
-from .conftest import chdir, package_dir
+from .conftest import chdir, in_git_repo_context, package_dir
 
 
 @pytest.mark.parametrize(
@@ -92,8 +92,9 @@ def test_user_args(package_user_args, tmp_path, monkeypatch):
         'install-args': ('cli-install',),
     }
 
-    mesonpy.build_sdist(tmp_path, config_settings)
-    mesonpy.build_wheel(tmp_path, config_settings)
+    with in_git_repo_context():
+        mesonpy.build_sdist(tmp_path, config_settings)
+        mesonpy.build_wheel(tmp_path, config_settings)
 
     # check that the right commands are executed, namely that 'meson
     # compile' is used on Windows rather than a 'ninja' direct

